biogeophys/FatesPlantHydraulicsMod.F90

```diff
@@ -1288,407 +1239,432 @@ end subroutine DeallocateHydrCohort
 
   ! =====================================================================================
 
-  subroutine InitHydrSites(sites,bc_in,numpft)
+  subroutine InitHydrSites(sites,bc_in)
 
-       ! Arguments
-       type(ed_site_type),intent(inout),target :: sites(:)
-       type(bc_in_type),intent(in)             :: bc_in(:)
-       integer,intent(in)                      :: numpft
+    ! Arguments
+    type(ed_site_type),intent(inout),target :: sites(:)
+    type(bc_in_type),intent(in)             :: bc_in(:)
 
-       ! Locals
-       integer :: nsites
-       integer :: s
-       type(ed_site_hydr_type),pointer :: csite_hydr
-       
+    ! Locals
+    integer :: nsites
+    integer :: s
+    integer :: j
+    integer :: jj
+    type(ed_site_hydr_type),pointer :: csite_hydr
 
-       if ( hlm_use_planthydro.eq.ifalse ) return
-       
-       ! Initialize any derived hydraulics parameters
-       call InitHydraulicsDerived(numpft)
+
+
+    if ( hlm_use_planthydro.eq.ifalse ) return
+
+    ! Initialize any derived hydraulics parameters
+
+    nsites = ubound(sites,1)
+    do s=1,nsites
+       allocate(csite_hydr)
+       sites(s)%si_hydr => csite_hydr
+       if ( bc_in(s)%nlevsoil > nlevsoi_hyd_max ) then
+          write(fates_log(),*) 'The host land model has defined soil with'
+          write(fates_log(),*) bc_in(s)%nlevsoil,' layers, for one of its columns.'
+          write(fates_log(),*) 'Fates-hydro temporary array spaces with size'
+          write(fates_log(),*) 'nlevsoi_hyd_max = ',nlevsoi_hyd_max,' must be larger'
+          write(fates_log(),*) 'see main/FatesHydraulicsMemMod.F90'
+          call endrun(msg=errMsg(sourcefile, __LINE__))
+       end if
+
+       ! Calculate the number of rhizosphere
+       ! layers used
+       if(ignore_layer1) then
+          csite_hydr%i_rhiz_t = 2
+          csite_hydr%i_rhiz_b = bc_in(s)%nlevsoil
+       else
+          csite_hydr%i_rhiz_t = 1
+          csite_hydr%i_rhiz_b = bc_in(s)%nlevsoil
+       end if
        
-       nsites = ubound(sites,1)
-       do s=1,nsites
-          allocate(csite_hydr)
-          sites(s)%si_hydr => csite_hydr
-          if ( bc_in(s)%nlevsoil > nlevsoi_hyd_max ) then
-             write(fates_log(),*) 'The host land model has defined soil with'
-             write(fates_log(),*) bc_in(s)%nlevsoil,' layers, for one of its columns.'
-             write(fates_log(),*) 'Fates-hydro temporary array spaces with size'
-             write(fates_log(),*) 'nlevsoi_hyd_max = ',nlevsoi_hyd_max,' must be larger'
-             write(fates_log(),*) 'see main/FatesHydraulicsMemMod.F90'
-             call endrun(msg=errMsg(sourcefile, __LINE__))
-          end if
-          sites(s)%si_hydr%nlevsoi_hyd = bc_in(s)%nlevsoil
-          call sites(s)%si_hydr%InitHydrSite()
+       csite_hydr%nlevrhiz = csite_hydr%i_rhiz_b-csite_hydr%i_rhiz_t+1
+       call sites(s)%si_hydr%InitHydrSite(numpft,nlevsclass)
```

Comment:
`numpft` is being used for this call but looks like it's not defined as an input to the call anymore.  I noticed this based on the call from the CTSM PR.  Should this be a locally defined variable?

Reply:
It is a global: https://github.com/NGEET/fates/blob/320a8fc0875e228df4133544f122c83844d8720e/biogeophys/FatesPlantHydraulicsMod.F90#L62

Reply:
doh :facepalm: .  I shoulda seen that.